### DISCUSSION TITLE
Allow Customization of Default Routing Prefix for Added HTTP Endpoints

### DISCUSSION THREAD

**Discussion by stephanjohnson** [IntentArchitect/Support#28] [General]
The default behavior for adding a new HTTP endpoint is to prefix the route with `api/`. This convention dates back to earlier practices when APIs were typically part of the same project as the UI. The `api/` prefix served as a clear distinction between the API endpoints and other parts of the application, providing organized routing, easier versioning, and even assisting certain authentication scenarios.

However, the landscape of software development has evolved, and it's now more common to have the backend and frontend separated into different projects, with each hosted independently. In this setup, APIs are often hosted on an `api.` sub-domain, leading to situations where the `api/` prefix on the route can appear redundant, such as `https://api.myproject.com/api/...`.

While the `api/` prefix still has its merits in certain scenarios, like namespace differentiation, versioning, and assisting certain load balancers or reverse proxies, its utility may diminish in cases where the backend is hosted on a dedicated `api.` subdomain.

To cater to these diverse modern practices, it might be beneficial to introduce an option allowing developers to customize the default routing prefix at the application level. This enhancement would enable developers to align their API configuration more closely with their specific deployment architectures and preferences, without being tied to a potentially redundant prefix.

**Comment by joelsteventurner** [comment]
Hi @stephanjohnson 

This sounds like a good enhancement, I have logged it as an enhancement on our back log.

Thanks for the suggestion.

**Comment by dandrejvv** [comment]
Hi @stephanjohnson 
There is a pre-release of `Intent.AspNetCore.Controllers` (5.4.0-pre.2) available where you can now adjust the Default Routing Prefix for your services. 
Once you've updated the module, head over to the Modules Settings page and locate the `API Settings` section and change the `API Routing Prefix` setting.
Please try it out and let us know what you think.
